packages/autocannon-ui-frontend/src/components/RunOptionsForm.jsx

```diff
@@ -0,0 +1,286 @@
+import React, { useState } from 'react'
+import { makeStyles } from '@material-ui/core/styles'
+import {
+  Grid,
+  Container,
+  TextField,
+  Tooltip,
+  FormControl,
+  InputLabel,
+  Select,
+  MenuItem,
+  Box,
+  Button
+} from '@material-ui/core'
+import T from 'prop-types'
+import HelpIcon from '@material-ui/icons/Help'
+
+import ProgressBar from './ProgressBar.jsx'
+import logo from '../assets/autocannon-banner.png'
+import { Alert } from '@material-ui/lab'
+
+const useStyles = makeStyles(theme => ({
+  form: {
+    width: '100%',
+    marginTop: theme.spacing(1)
+  },
+  logo: {
+    maxWidth: '70%',
+    display: 'block',
+    margin: 'auto',
+    padding: '10px'
+  },
+  helpIcon: {
+    fontSize: 'medium',
+    cursor: 'default'
+  },
+  runButton: {
+    borderRadius: '30px'
+  },
+  actionSection: {
+    padding: '20px 0px'
+  },
+  progressBar: {
+    padding: '30px'
+  },
+  errorSection: {
+    padding: '20px',
+    borderRadius: '5px'
+  }
+}))
+
+export default function RunOptionsForm(props) {
+  const classes = useStyles()
+
+  const [options, setOptions] = useState({
+    url: 'https://google.com',
+    connections: 10,
+    pipelining: 1,
+    duration: 10,
+    method: 'GET'
+  })
+  const [progress, setProgress] = useState(0)
+  const [isTestRunning, setIsTestRunning] = useState(false)
+  const [request, setRequest] = useState()
+  const [errorMessage, setErrorMessage] = useState()
+  const [invalid, setInvalid] = useState(false)
+  const [validationResults, setValidationResults] = useState({})
+
+  function onOptionChange(option, event) {
+    if (event.target.required && !event.target.value) {
+      setValidationResults(results => ({ ...results, [event.target.id]: true }))
+      setInvalid(true)
+    } else {
+      setValidationResults(results => {
+        delete results[event.target.id]
+        return results
+      })
+      setInvalid(Object.keys(validationResults).length > 0)
+    }
+    setOptions(o => ({ ...o, [option]: event.target.value }))
+  }
+
+  const runButtonHandler = () => {
+    setIsTestRunning(true)
+    setProgress(0)
+    setErrorMessage()
+
+    const controller = new AbortController()
+    const promise = runTest(controller.signal)
+    promise.cancel = () => controller.abort()
+    setRequest(promise)
+    promise
+      .then(result => {
+        props.onNewResults(result)
+      })
+      .catch(e => {
+        if (e.name !== 'AbortError') {
+          setErrorMessage(e.message)
+        }
+      })
+      .finally(() => {
+        setIsTestRunning(false)
+      })
```

Comment:
let's use async/await for consistency with the rest of the code. let's avoid mixing async/await with promise native methods in general